shoes-core/lib/shoes/download.rb

```diff
@@ -140,9 +142,23 @@ def save_to_file(file_path)
       open(file_path, 'wb') { |fw| fw.print @response.body }
     end
 
-    def download_started(content_length)
-      @content_length = content_length
-      @started        = true
+    def download_started(response)
+      unless @started
```

Comment:
Yeah, it's slightly cryptic this way. I had a version where `HttpWrapper` had tons of callbacks and it wasn't pretty, but I think doing one for started might work. Will check it and see.